### PR TITLE
Remove config key order check

### LINKED ISSUE #1471: Receiving the `WARNING: top-level config key bases must be defined before environments in ...` despite it being so
### Operating system

Ubuntu 22.04.4 LTS

### Helmfile Version

v0.162.0

### Helm Version

v3.14.4

### Bug description

Helmfile issues a warning, as seen in the error message, when a YAML document defining `environment` is followed by another YAML document (rendered later) that defines `bases`, even though other `bases` were properly defined before `environments`.

### Example helmfile.yaml

```yaml
bases: []
---
environments:
  "{{ .Environment.Name }}": {}
---
bases: []  # This is intentional!
---
releases:
  - name: test
    chart: .
```
> **NOTE:** The use of such a construct (the second `bases` after `environments`) is intentional and is perfectly correct if you know what you are doing. I personally believe that the generated warning, though well-intentioned, is too invasive. Moreover, it cannot be suppressed (ideally, it could be suppressed either via a CLI flag or through `helmfileDefaults` within the configuration). We employ quite complex configurations, layered from several `helmfile.yaml` files and additional logic. We understand very well why and how we do it, and it works perfectly. On the other hand, users who deploy our helmfile project are confused and ask what it means. Since they do not use any intermediary scripts but directly use `helmfile apply git::https://...`, there is no way to get rid of it.

#### helmfile.yaml
```yaml
bases: []
---
environments:
  "{{ .Environment.Name }}": {}
---
releases:
  - name: test
    chart: .
```
#### ✅ Helmfile Output - Correct:
```console
$ helmfile list
NAME    NAMESPACE       ENABLED INSTALLED       LABELS  CHART   VERSION
test                    true    true
```

#### helmfile.yaml
```yaml
environments:
  "{{ .Environment.Name }}": {}
---
bases: []
---
releases:
  - name: test
    chart: .
```
#### ✅ Helmfile Output - Correct:
```console
$ helmfile list
WARNING: top-level config key bases must be defined before environments in helmfile.yaml
NAME    NAMESPACE       ENABLED INSTALLED       LABELS  CHART   VERSION
test                    true    true
```

#### helmfile.yaml
```yaml
bases: []
---
environments:
  "{{ .Environment.Name }}": {}
---
bases: []
---
releases:
  - name: test
    chart: .
```
#### ❌ Helmfile Output - Incorrect:
```console
$ helmfile list
WARNING: top-level config key bases must be defined before environments in helmfile.yaml
NAME    NAMESPACE       ENABLED INSTALLED       LABELS  CHART   VERSION
test                    true    true
```

#### Helmfile Output - Expected Output:
```console
$ helmfile list
NAME    NAMESPACE       ENABLED INSTALLED       LABELS  CHART   VERSION
test                    true    true
```

### Steps to reproduce

see previous examples

### Working Helmfile Version

version before this Warning was introduced

### Relevant discussion

_No response_

### PR DESCRIPTION
**Pull request by drivelikebrazil**
Removes the top-level config key order check introduced in #710.

The check introduced in the above PR adds restrictions that prevent valid advanced use cases:

```
bases:
  - baseline-config.yaml
---
environment:
  values:
    - myValues.yaml
---
bases:
  - {{ .Values.someEnvironmentVal }}-defaults.yaml.gotmpl
---
releases: ...
```

In both v1 and v0 mode, this is valid structure that will template correctly without this policy. I can understand the intent for the rule to help weed out cases where someone tries to use an environment variable *before* it's defined, but it shouldn't prevent users who understand the ordering rules from fully leveraging helmfile's layering capabilities.

Resolves #1471

### CODE REVIEW

File: pkg/policy/checker.go [hunk 1]
```diff
@@ -15,11 +15,6 @@ var (
 	EnvironmentsAndReleasesWithinSameYamlPartErr = errors.New("environments and releases cannot be defined within the same YAML part. Use --- to extract the environments into a dedicated part")
 	topConfigKeysRegex                           = regexp.MustCompile(`^[a-zA-Z]+:`)
 	separatorRegex                               = regexp.MustCompile(`^--- *$`)
-	topkeysPriority                              = map[string]int{
```
Comment (jouve) [inline pkg/policy/checker.go:18]: maybe we can keep bases & environments with the same priority ?

Comment (yxxhero) [reply pkg/policy/checker.go:18]: yeah. please have a try.

Comment (drivelikebrazil) [reply pkg/policy/checker.go:18]: Implemented as requested

**Review by yxxhero — APPROVED** [review]

**Comment by yxxhero** [comment]
@drivelikebrazil thanks so much.

**Pull request merged**